packages/fluentui/react-northstar/src/themes/teams/components/Chat/chatMessageStyles.ts

```diff
@@ -39,44 +39,76 @@ export const chatMessageStyles: ComponentSlotStylesPrepared<ChatMessageStylesPro
       paddingTop: pxToRem(8),
       paddingBottom: pxToRem(10),
 
-      borderRadius: v.borderRadius,
-      border: v.border,
-      outline: 0,
-
-      color: v.color,
       backgroundColor: p.mine ? v.backgroundColorMine : v.backgroundColor,
+    }),
+
+    borderRadius: v.borderRadius,
+    border: p.compact ? v.compactBorder : v.border,
+    outline: 0,
+
+    color: v.color,
+
+    wordBreak: 'break-word',
+    wordWrap: 'break-word',
+
+    ...((v.hasMention || v.isImportant) && {
+      '::before': {
+        backgroundColor: v.hasMention ? v.hasMentionColor : v.isImportantColor,
+        position: 'absolute',
+        ...(p.compact
+          ? {
+              borderRadius: pxToRem(2),
+              height: `calc(100% - ${v.paddingCompact} - ${v.paddingCompact})`,
+              left: pxToRem(-56),
+              top: v.paddingCompact,
+              width: pxToRem(2),
+            }
+          : {
+              borderBottomLeftRadius: 'inherit',
+              borderTopLeftRadius: 'inherit',
+              height: '100%',
+              left: '0',
+              top: '0',
+              width: pxToRem(3),
+            }),
+      },
+    }),
+
+    ...getBorderFocusStyles({ borderRadius: 'inherit', variables: siteVariables }),
 
-      wordBreak: 'break-word',
-      wordWrap: 'break-word',
-
-      ...((v.hasMention || v.isImportant) && {
-        [`& .${chatMessageSlotClassNames.bar}`]: {
-          backgroundColor: v.hasMention ? v.hasMentionColor : v.isImportantColor,
-          height: '100%',
-          left: '0',
-          position: 'absolute',
-          top: '0',
-          width: pxToRem(3),
-          borderBottomLeftRadius: 'inherit',
-          borderTopLeftRadius: 'inherit',
+    ...(p.compact && {
```

Comment:
I think we need to consider a different folder structure for styles:
* themes/teams/components/Chat
  * compact
    * chatMessageStyles.ts // factories for slot styles
    * ...
  * standard
    * chatMessageStyles.ts // factories for slot styles
    * ...
   * chatMessageStyles.ts // toggle the factories based on `p.layout` (i.e `p.compact`)

I don't mind duplicating at this point, but looking at the styles, `p.compact` is not spread everywhere in the following ways:
* ternary condition for values `value: p.compact ? x : y`
* ternary condition for object spread `p.compact ? {} : {}`
* conditional object spread `p.compact && {}`

Reply:
Followed your second file-structure suggestion to separate `compact` and `comfy` styles.